### ISSUE TITLE
transducer data process

### ISSUE THREAD

**Issue by alsm168** [Z-yq/TensorflowASR#1]
hello:
I want to train rnn-transducer, where to use prepand_blank() of TextFeaturizer? thank you.

**Comment by Z-yq** [comment]
hi,
you can change the code in utils/text_featurizers.py to meet you needs.


line 112 for transducer:


feats=[self.start]+[self.token_to_index[token] for token in tokens]+[self.stop]
self.start default is token 's' 

self.stop default is token&nbsp; '/s'


you&nbsp; can set&nbsp; 'blank_at_zeros = False' that in am_data.yml ,or modify their value.


by the way:
wrap_rnnt_loss is more stable.

------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Z-yq/TensorflowASR"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年9月25日(星期五) 下午3:57
收件人:&nbsp;"Z-yq/TensorflowASR"<TensorflowASR@noreply.github.com&gt;;
抄送:&nbsp;"Subscribed"<subscribed@noreply.github.com&gt;;
主题:&nbsp;[Z-yq/TensorflowASR] transducer data process (#1)





 
hello:
 I want to train rnn-transducer, where to use prepand_blank(0 of TextFeaturizer? thank you.

**Comment by alsm168** [comment]
> hi, you can change the code in utils/text_featurizers.py to meet you needs. line 112 for transducer: feats=[self.start]+[self.token_to_index[token] for token in tokens]+[self.stop] self.start default is token 's' self.stop default is token&nbsp; '/s' you&nbsp; can set&nbsp; 'blank_at_zeros = False' that in am_data.yml ,or modify their value. by the way: wrap_rnnt_loss is more stable.
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "Z-yq/TensorflowASR" <notifications@github.com&gt;; 发送时间:&nbsp;2020年9月25日(星期五) 下午3:57 收件人:&nbsp;"Z-yq/TensorflowASR"<TensorflowASR@noreply.github.com&gt;; 抄送:&nbsp;"Subscribed"<subscribed@noreply.github.com&gt;; 主题:&nbsp;[Z-yq/TensorflowASR] transducer data process (#1) hello: I want to train rnn-transducer, where to use prepand_blank(0 of TextFeaturizer? thank you. — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe.

thanks for your reply. to use rnnt-loss, blank should be place as 's blank a b c /s'  or  'blank  s  a b c /s' ?

**Comment by Z-yq** [comment]
sorry , I didn't reply in time。


all right ， we should not to insert blank into label, this step will be done in loss function.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Z-yq/TensorflowASR"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年9月25日(星期五) 晚上6:41
收件人:&nbsp;"Z-yq/TensorflowASR"<TensorflowASR@noreply.github.com&gt;;
抄送:&nbsp;"　Ｄｅｍｏｎ丶"<641242921@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题:&nbsp;Re: [Z-yq/TensorflowASR] transducer data process (#1)





  
hi, you can change the code in utils/text_featurizers.py to meet you needs. line 112 for transducer: feats=[self.start]+[self.token_to_index[token] for token in tokens]+[self.stop] self.start default is token 's' self.stop default is token&nbsp; '/s' you&nbsp; can set&nbsp; 'blank_at_zeros = False' that in am_data.yml ,or modify their value. by the way: wrap_rnnt_loss is more stable.
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "Z-yq/TensorflowASR" <notifications@github.com&gt;; 发送时间:&nbsp;2020年9月25日(星期五) 下午3:57 收件人:&nbsp;"Z-yq/TensorflowASR"<TensorflowASR@noreply.github.com&gt;; 抄送:&nbsp;"Subscribed"<subscribed@noreply.github.com&gt;; 主题:&nbsp;[Z-yq/TensorflowASR] transducer data process (#1) hello: I want to train rnn-transducer, where to use prepand_blank(0 of TextFeaturizer? thank you. — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe.
  
thanks for your reply. to use rnnt-loss, blank should be place as 'a b c '  or  ' a b c ' ?

**Comment by alsm168** [comment]
thank you，I got it.😂



---Original---
